pkg/fleet/installer/setup/djm/dataproc.go

```diff
@@ -0,0 +1,105 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package djm contains data-jobs-monitoring installation logic
+package djm
+
+import (
+	"cloud.google.com/go/compute/metadata"
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/DataDog/datadog-agent/pkg/fleet/installer/setup/common"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	dataprocInjectorVersion   = "0.26.0-1"
+	dataprocJavaTracerVersion = "1.42.2-1"
+	dataprocAgentVersion      = "7.58.2-1"
+)
+
+var (
+	tracerEnvConfigDataproc = []common.InjectTracerConfigEnvVar{
+		{
+			Key:   "DD_DATA_JOBS_ENABLED",
+			Value: "true",
+		},
+		{
+			Key:   "DD_INTEGRATIONS_ENABLED",
+			Value: "false",
+		},
+		{
+			Key:   "DD_DATA_JOBS_COMMAND_PATTERN",
+			Value: ".*org.apache.spark.deploy.*",
+		},
+		{
+			Key:   "DD_SPARK_APP_NAME_AS_SERVICE",
+			Value: "true",
+		},
+	}
+)
+
+// SetupDataproc sets up the DJM environment on Dataproc
+func SetupDataproc(s *common.Setup) error {
+
+	metadataClient := metadata.NewClient(nil)
```

Comment:
Nit: as this basically wraps http calls to the metadata endpoints, i wonder if wrapping the transport could let us trace these calls
Like so: https://github.com/DataDog/datadog-agent/blob/179220f0b172ccdfcff02346de3102ee84476939/pkg/fleet/installer/oci/download.go#L244

That way we can know if customers have errors and if we need retries